### ISSUE TITLE
Overly agressive constness check on comptime copied slices

### ISSUE THREAD

**Issue by N00byEdge** [ziglang/zig#8644]
```
  comptime var x: []u8 = &[_]u8{};
  x = x ++ "a";
```
Fails with the following error:
```
./example.zig:3:9: error: expected type '[]u8', found '*const [1:0]u8'
  x = x ++ "a";
        ^
```
Godbolt: https://gcc.godbolt.org/z/he94PY3bx
Should probably be allowed since this is a value copy, not a reference

**Comment by N00byEdge** [comment]
Workaround:
```
  comptime var x: []const u8 = &[_]u8{};
  x = x ++ "a";
```

**Comment by RetroDev256** [comment]
This still persists in master - not a stage1 issue alone.

**Comment by andrewrk** [comment]
Fixed in cfdb001a8f0353466a4445cc046341ce9cc43fc5
